File: Java_template/NewRoads.java
public class NewRoads {
    public NewRoads(int n) {
        // TODO
    }

    public void addRoad(int a, int b, int x) {
        // TODO
    }

    public int minCost() {
        // TODO
    }

    public static void main(String[] args) {
        NewRoads n = new NewRoads(4);
        n.addRoad(1,2,2);
        n.addRoad(1,3,5);
        System.out.println(n.minCost()); // -1
        n.addRoad(3,4,4);
        System.out.println(n.minCost()); // 11
        n.addRoad(2,3,1);
        System.out.println(n.minCost()); // 7
    }
}

File: newroads.py
class NewRoads:
    def __init__(self,n):
        # TODO

    def add_road(self,a,b,x):
        # TODO

    def min_cost(self):
        # TODO

if __name__ == "__main__":
    n = NewRoads(4)
    n.add_road(1,2,2)
    n.add_road(1,3,5)
    print(n.min_cost()) # -1
    n.add_road(3,4,4)
    print(n.min_cost()) # 11
    n.add_road(2,3,1)
    print(n.min_cost()) # 7
